src/components/forms/DomainSettings/LogoRow.tsx

```diff
@@ -0,0 +1,129 @@
+import ValidationInput from '@src/components/inputs/text/ValidationInput/ValidationInput';
+import ArweaveID, {
+  ArweaveIdTypes,
+} from '@src/components/layout/ArweaveID/ArweaveID';
+import ConfirmTransactionModal from '@src/components/modals/ConfirmTransactionModal/ConfirmTransactionModal';
+import { ArweaveTransactionID } from '@src/services/arweave/ArweaveTransactionID';
+import { useGlobalState } from '@src/state/contexts/GlobalState';
+import {
+  ANT_INTERACTION_TYPES,
+  ContractInteraction,
+  VALIDATION_INPUT_TYPES,
+} from '@src/types';
+import { isArweaveTransactionID } from '@src/utils';
+import { ARNS_TX_ID_ENTRY_REGEX } from '@src/utils/constants';
+import eventEmitter from '@src/utils/events';
+import { Skeleton } from 'antd';
+import { useEffect, useState } from 'react';
+
+import DomainSettingsRow from './DomainSettingsRow';
+
+export default function LogoRow({
+  logoTxId,
+  confirm,
+  editable = false,
+}: {
+  logoTxId?: string;
+  confirm: (logo: string) => Promise<ContractInteraction>;
+  editable?: boolean;
+}) {
+  const [editing, setEditing] = useState<boolean>(false);
+  const [newLogoTxId, setNewLogoTxId] = useState<string>(logoTxId ?? '');
+  const [{ arweaveDataProvider }] = useGlobalState();
+  const [showModal, setShowModal] = useState<boolean>(false);
+
+  useEffect(() => {
+    setNewLogoTxId(logoTxId ?? '');
+  }, [logoTxId]);
+
+  async function handleSave(transactionId: string) {
+    try {
+      await confirm(transactionId);
+    } catch (error) {
+      eventEmitter.emit('error', error);
+    } finally {
+      setEditing(false);
+      setNewLogoTxId(logoTxId ?? '');
```

Comment:
Double-checking: If confirm fails, this is still setting logoTxId. Is that the correct behavior?

Reply:
yeah we reset to the original txid